api/middleware/middleware.go

```diff
@@ -215,3 +219,32 @@ func ForwardToServer(c *gin.Context, fromID node.ID, toAddr string) {
 		return
 	}
 }
+
+// KeyspaceCheckerMiddleware check if the request keyspace is valid
+func KeyspaceCheckerMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		keyspace := c.Query(api.APIOpVarKeyspace)
+		if keyspace != "" && keyspace != common.DefaultKeyspace {
+			pdAPIClient := appcontext.GetService[pdutil.PDAPIClient](appcontext.PDAPIClient)
+
+			meta, err := pdAPIClient.LoadKeyspace(c.Request.Context(), keyspace)
+			if errors.IsKeyspaceNotExistError(err) {
+				c.IndentedJSON(http.StatusBadRequest, errors.ErrAPIInvalidParam)
+				c.Abort()
+				return
+			} else if err != nil {
```

Comment:
```suggestion
			}
     if err != nil {
```

Reply:
This error check can be simplified, check err != nil first, then heck errors.IsKeyspaceNotExistError(err)